packages/core/src/event-bus/event-bus.ts

```diff
@@ -123,7 +123,7 @@ export class EventBus implements OnModuleDestroy {
             return event;
         }
 
-        return this.transactionSubscriber.awaitRelease(transactionManager.queryRunner).then(() => {
+        return this.transactionSubscriber.awaitCommit(transactionManager.queryRunner).then(() => {
```

Comment:
I'm wondering whether there is an issue here - if the transaction fails and there is no commit, we will have a promise that never resolves right?

Reply:
Ah, sorry, I though you are about code before changes. I see your point, yes, it seems we will have a promise that will never resolve. I'm not sure whether this can be considered as a memory leak